Add callback function for poll events to client api

andreas-bok-sociomantic opened the pull request:
This PR addresses the issue raised, https://answers.launchpad.net/libdrizzle/+question/233212, 
in which the removal of `event_watch_fn` meant that a client could no longer 
suscribe to and get notified about poll events

andreas-bok-sociomantic committed:
Implement event_watch_fn

Add member variable and function to drizzle_st struct
Add parameters with default values to drizzle_st
constructor

Add function to set function and context on
drizzle_st struct

Handle event_watch_fn when setting poll event flags

Test event_watch callback function when event
flags are set using `drizzle_set_events` and
`drizzle_set_revents`

If return value from `event_watch_fn` is not
`DRIZZLE_RETURN_OK`, the connection to drizzle
is closed.

The changes are adapted from `libdrizzle 1.0` which
implements the event_watch_fn callback function

mathias-baumann-sociomantic commented:
LGTM

ben-palmer-sociomantic commented:
Generally LGTM but perhaps it would be best to add an entry to the release notes about this change.

andreas-bok-sociomantic commented:
Agree on https://github.com/sociomantic/libdrizzle-redux/pull/5#issuecomment-162860691 
though should the first "release" simply be the officieal `libdrizzle redux v5.1.4`

leandro-lucarella-sociomantic commented:
You can assume the official v5.1.4 was already released and call next release v5.2.0 for example, so the release notes includes all changes between the official v5.1.4 and v.5.2.0.

ben-palmer-sociomantic commented:
ping @andreas-bok-sociomantic once this has the release notes I am happy to rebase it.

andreas-bok-sociomantic commented:
@ben-palmer-sociomantic, added release notes.

andreas-bok-sociomantic commented:
Added documentation for `drizzle_set_event_watch_fn` to the libdrizzle manual

ben-palmer-sociomantic commented on docs/api/connection.rst:
```diff
@@ -194,7 +203,7 @@ Functions
    :param options: The options object to get the value from
    :returns: The state of the auth plugin option
-   
+
```
This whitespace change should be in its own commit.

ben-palmer-sociomantic commented:
Apart from my comment about the whitespace seems to be looking good, can you squash together any commits that need it before I rebase?

andreas-bok-sociomantic committed:
Add RELEASE NOTES and documentation in manual

**Add RELEASE NOTES for in-development version**
- describes dependencies and the added support
  for custom callback function

**Update rst documentation to cover**
- `drizzle_event_watch_fn`
- `drizzle_set_event_watch_fn`

andreas-bok-sociomantic commented:
Squashed commits. Left whitespace fix out of this commit to keep the history more clean.

ben-palmer-sociomantic commented:
LGTM, @mathias-baumann-sociomantic any other comments?

mathias-baumann-sociomantic commented:
No comments.

ben-palmer-sociomantic commented:
This pull request has been rebased via `git hub pull rebase`. Original pull request HEAD was 7ad230ede5dab4b56fe066be384a966a28553447, new (rebased) HEAD is 0034b383f72f15c340c94a8b6ed8e1ebd0d12715
